### PR TITLE
Add `examples/language_modeling/run_clm_no_trainer.py`

### PR DESCRIPTION
**Pull request by hemildesai**
# What does this PR do?

This PR adds an example of finetuning a Causal Language Model (without using Trainer) to show the functionalities of the new accelerate library.

## Who can review?

@sgugger

### CODE REVIEW

File: examples/language-modeling/run_clm_no_trainer.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+)
+
+
+logger = logging.getLogger(__name__)
+MODEL_CONFIG_CLASSES = list(MODEL_MAPPING.keys())
+MODEL_TYPES = tuple(conf.model_type for conf in MODEL_CONFIG_CLASSES)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Finetune a transformers model on a text classification task")
```
Comment (sgugger) [inline examples/language-modeling/run_clm_no_trainer.py:59]:
```suggestion
    parser = argparse.ArgumentParser(description="Finetune a transformers model on a causal language modeling task")
```

**Review by sgugger — APPROVED** [review]
Thanks a lot for adding this script! It works perfectly on my local tests.

You should now add instructions in the [README](https://github.com/huggingface/transformers/blob/master/examples/language-modeling/README.md) on how to use this script and the `run_mlm_no_trainer` script you introduced before (either in this PR or a new one, as you prefer).

**Comment by hemildesai** [comment]
@sgugger Awesome, I'll create a new PR for it.

**Comment by sgugger** [comment]
Merging this one then, thanks again!

**Pull request merged**